packages/bvaughn-architecture-demo/components/sources/SourceExplorer.tsx

```diff
@@ -0,0 +1,84 @@
+import { useContext, useMemo } from "react";
+import { ReplayClientContext } from "shared/client/ReplayClientContext";
+
+// import Comment from "./Comment";
+import styles from "./SourceExplorer.module.css";
+import { getSourcesToDisplay } from "@bvaughn/src/suspense/SourcesCache";
+
+// TODO Dim comments that are outside of the focus window
+
+export default function SourceExplorer() {
+  const client = useContext(ReplayClientContext);
+  const sources = getSourcesToDisplay(client);
+
+  console.log("!!", sources);
+  const sourceTree = useMemo(() => {
+    const tree = [];
+    let protocol: string | null = null;
+    let origin: string | null = null;
+    sources.forEach(source => {
+      if (!source.url) {
+        return;
+      }
+      const url = new URL(source.url);
+      if (url.protocol == "replay-content:") {
+        return;
+      }
+      if (
+        url.protocol != "replay-content:" &&
+        url.protocol != "https:" &&
+        url.protocol !== protocol
+      ) {
+        tree.push({
+          type: "protocol",
+          protocol: url.protocol,
+        });
+        protocol = url.protocol;
+      }
+      if (url.origin != null && url.origin !== origin) {
+        tree.push({
+          type: "origin",
+          origin: url.origin,
+        });
+        origin = url.origin;
+      }
+      if (url.pathname) {
+        tree.push({
+          type: "source",
+          source: source,
+          path: url.pathname,
+        });
+      }
+    });
+    return tree;
+  }, [sources]);
+
+  console.log(`sourceTree`, sourceTree);
+  return (
+    <div className={styles.List}>
+      <div className={styles.Header}>Sources</div>
+      <div className={styles.SourceTree}>
+        {sourceTree.map(node => {
+          if (node.type == "protocol") {
+            return <div key={node.protocol}>{node.protocol}</div>;
+          }
+
+          if (node.type == "origin") {
+            return (
+              <div key={node.origin} className={styles.Origin}>
+                {node.origin}
+              </div>
+            );
+          }
+          if (node.type == "source") {
+            return (
+              <div key={node.source.sourceId} className={styles.Source}>
+                {node.path}
+              </div>
+            );
+          }
+        })}
+      </div>
```

Comment:
I'm not really sure how our current app takes a source path like `"file:///Users/bvaughn/Desktop/temp-error-stack.html"` and turns it into this:
![Screen Shot 2022-07-05 at 8 37 30 AM](https://user-images.githubusercontent.com/29597/177333150-38792be9-6dab-4f4c-a144-8ca6c9125d12.png)

Why do we split "Users" and "bvaughn/Desktop" into separate chunks? Why do we split at all? and if we do, why 2 chunks instead of 3?